nuxeo-services/nuxeo-usermapper/src/main/java/org/nuxeo/usermapper/extension/AbstractUserMapper.java

```diff
@@ -151,7 +152,14 @@ protected DocumentModel createPrincipal(Map<String, Serializable> attributes) {
         UserManager userManager = getUserManager();
         DocumentModel userModel = userManager.getBareUserModel();
         userModel.getDataModel(userManager.getUserSchemaName()).setMap((Map) attributes);
-        return userManager.createUser(userModel);
+        DocumentModel userDoc;
+        try {
+            userDoc = Framework.doPrivileged(() -> userManager.createUser(userModel));
+        } catch (NuxeoException e) {
+            log.error("Error while creating user in UserManager", e);
+            return null;
+        }
+        return userDoc;
```

Comment:
You can put the `return` inside the `try`:
 
         try {
             return Framework.doPrivileged(() -> userManager.createUser(userModel));
         } ...